front50-gcs/src/main/java/com/netflix/spinnaker/front50/config/GcsConfig.java

```diff
@@ -0,0 +1,174 @@
+  private String applicationVersion;
+
+  private void ensureBucket(Storage storage) throws IOException {
+    Logger log = LoggerFactory.getLogger(getClass());
+    try {
+        Bucket bkt = storage.buckets().get(bucket).execute();
+        boolean has_versioning = bkt.getVersioning().getEnabled();
+        log.info("Bucket versioning is {}.",
+                 has_versioning ? "enabled" : "DISABLED");
+    } catch (HttpResponseException e) {
+      if (e.getStatusCode() == 404) {
+          log.warn("Bucket {} does not exist. Create it in project={}",
+                   bucket, project);
+          Bucket.Versioning versioning = new Bucket.Versioning().setEnabled(true);
+          Bucket spec = new Bucket().setName(bucket).setVersioning(versioning);
+          try {
+              storage.buckets().insert(project, spec).execute();
+          } catch (IOException e2) {
+              log.error("Could not create bucket={} in project={}: {}",
+                        bucket, project, e2);
+          }
+      } else {
+          log.error("Could not get bucket={}: {}", bucket, e);
+          return;
+      }
+    }
+      
+  }
+
+  @Bean
+  public Storage googleCloudStorageClient() {
+      try {
+        HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+        GoogleCredential credential;
+        Logger log = LoggerFactory.getLogger(getClass());
+
+        if (!jsonPath.isEmpty()) {
+            FileInputStream credentialStream = new FileInputStream(jsonPath);
+            credential = GoogleCredential.fromStream(credentialStream, httpTransport, jsonFactory)
+                                         .createScoped(Collections.singleton(StorageScopes.DEVSTORAGE_FULL_CONTROL));
+          log.info("Loaded credentials from from " + jsonPath);
+        } else {
+            log.info("spinnaker.gcs.enabled without providers.google.primaryCredentials.jsonPath. Using default application credentials. Using default credentials.");
+            credential = GoogleCredential.getApplicationDefault();
+        }
+
+        String applicationName = "Spinnaker-front50/" + applicationVersion;
+        Storage storage = new Storage.Builder(httpTransport, jsonFactory, credential)
+                .setApplicationName(applicationName)
+                .build();
+        try {
+            ensureBucket(storage);
+        } catch (IOException e) {
+            // ignore
+        }
+        return storage;
+      } catch (java.io.IOException|java.security.GeneralSecurityException e) {
+          throw new IllegalStateException(e);
+      }
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(RestTemplate.class)
+  public RestTemplate restTemplate() {
+    return new RestTemplate();
+  }
+
+  @Bean
+  public ObjectMapper gcsObjectMapper() {
+    return new ObjectMapper();
+  }
+
+  @Bean
+  @ConditionalOnExpression("${spinnaker.gcs.enabled:false}")
```

Comment:
Probably worth while to just put this `@ConditionalOnExpression` at the class-level.

Otherwise you're going to get some beans created when GCS is not enabled.